storage/series/buffer.go

```diff
@@ -337,6 +321,22 @@ func (b *dbBufferBucket) resetTo(start time.Time) {
 	b.drained = false
 }
 
+func (b *dbBufferBucket) canRead() bool {
+	return !b.drained && !b.empty
+}
+
+func (b *dbBufferBucket) needsReset(targetStart time.Time) bool {
+	return !b.start.Equal(targetStart)
+}
+
+func (b *dbBufferBucket) needsDrain(now time.Time, targetStart time.Time) bool {
+	retentionOpts := b.opts.RetentionOptions()
+	blockSize := retentionOpts.BlockSize()
+	bufferPast := retentionOpts.BufferPast()
```

Comment:
That adds 8 bytes per bucket, which can sum up to a few hundred MBs. These should be three fast memory accesses per bucket and likely cached, which should be on the order of tens of nanoseconds, which sums up to tens of milliseconds across all series. Since the tick always takes a minute and sleeps in between to make up the slack, this likely means we'll just sleep a bit less and have no real impact on the tick time.